src/algebra/big_operators/fin.lean

```diff
@@ -114,6 +114,46 @@ lemma prod_filter_succ_lt {M : Type*} [comm_monoid M] {n : ℕ} (j : fin n) (v :
     ∏ j in univ.filter (λ i, j < i), v j.succ :=
 by rw [univ_filter_succ_lt, finset.prod_map, rel_embedding.coe_fn_to_embedding, coe_succ_embedding]
 
+@[to_additive]
+lemma prod_congr' {M : Type*} [comm_monoid M] {a b : ℕ} (f : fin b → M) (h : a = b) :
+  ∏ (i : fin a), f (cast h i) = ∏ (i : fin b), f i :=
+by { subst h, congr, ext, congr, ext, rw coe_cast, }
+
+@[to_additive]
+lemma prod_univ_add {M : Type*} [comm_monoid M] {a b : ℕ} (f : fin (a+b) → M) :
```

Comment:
I think it might be helpful to factor this through:
```lean
@[to_additive]
lemma prod_of_sum {ι κ M : Type*} [fintype ι] [fintype κ] [comm_monoid M] (f : ι ⊕ κ → M) :
  ∏ (i : ι ⊕ κ), f i = (∏ (i : ι), f (sum.inl i)) * (∏ (i : κ), f (sum.inr i)) :=
sorry
```
and then use https://leanprover-community.github.io/mathlib_docs/logic/equiv/fin.html#fin_sum_fin_equiv

Reply:
Thanks. I have used `prod_sum_elim` in the last commit.